examples/module1/lesson1/solver/App.tsx

```diff
@@ -16,43 +29,32 @@ const App = () => {
         <input
           type="number"
           className="rounded-md shadow-md p-4"
-          value={numA}
-          onChange={(e) => setNumA(parseFloat(e.target.value))}
+          value={operandLeft}
+          onChange={(e) => handleInputChange(setOperandLeft, e.target.value)}
         />
         <input
           type="number"
           className="rounded-md shadow-md p-4"
-          value={numB}
-          onChange={(e) => setNumB(parseFloat(e.target.value))}
+          value={operandRight}
+          onChange={(e) => handleInputChange(setOperandRight, e.target.value)}
         />
       </div>
       <div className="grid grid-cols-4 gap-x-4 my-4">
-        <button
-          className="bg-blue-200 px-2 py-4 text-lg hover:bg-blue-500 hover:text-white rounded-md"
-          onClick={() => doWork(f1)}
-        >
-          +
-        </button>
-        <button
-          className="bg-blue-200 px-2 py-4 text-lg hover:bg-blue-500 hover:text-white rounded-md"
-          onClick={() => doWork(f2)}
-        >
-          -
-        </button>
-        <button
-          className="bg-blue-200 px-2 py-4 text-lg hover:bg-blue-500 hover:text-white rounded-md"
-          onClick={() => doWork(f3)}
-        >
-          *
-        </button>
-        <button
-          className="bg-blue-200 px-2 py-4 text-lg hover:bg-blue-500 hover:text-white rounded-md"
-          onClick={() => doWork(f4)}
-        >
-          /
-        </button>
+        {operations.map((operation, index) => (
+          <Button
```

Comment:
czy tutaj potrzebujemy z tego tworzyc nowy komponent? jak dla mnie jest to tylko utrudnienie bo teraz to wygląda to na coś "skomplikowanego" a było mega proste na początku, staraj się robić rzeczy najprosciej jak sie da, nie dodawaj kolejnych złożoności do prostych rzeczy